Tests/AccessibilityTests/RedundantPublicAccessibilityTest.swift

```diff
@@ -1,8 +1,9 @@
-import XCTest
-@testable import TestShared
 @testable import PeripheryKit
+@testable import TestShared
+import XCTest
 
-class RedundantPublicAccessibilityTest: SourceGraphTestCase {
+// swiftlint:disable:next balanced_xctest_lifecycle
```

Comment:
So the classic case where they do/should is 

```swift
final class SomeTests: XCTestCase {
    // swiftlint:disable:next implicitly_unwrapped_optional
    private var someString: String!

    override func setUp() {
        super.setUp()
        someString = "Hello"
    }
}
```

without a 

```swift
    override func tearDown() {
       someString = nil
       super.tearDown()
    }
```

for ever test function, you'll end up with another copy of "Hello" - which is pretty minor, but in the aggregate it adds up, and if you keep the unit tests memory clean, you can more easily spot memory leaks in the app code (breakpointing on the last unit test and hitting the memory debugger is a nice way to do this).

Of course this is all to taste, the nature of the codebase, whether you're willing to put up with some `disables`, and what your threshold is for that.

Totally up to you whether you want to keep this on or not ...

Reply:
Let's just keep it for now since there are only a couple of (valid) violations. I really need to open-source the test leak detector I implemented at Reddit, which would negate the need for this rule.